src/org/ohdsi/usagi/ui/actions/ExitAction.java

```diff
@@ -31,7 +32,10 @@ public ExitAction() {
 
 	@Override
 	public void actionPerformed(ActionEvent arg0) {
-		System.exit(0);
+		if (UsagiDialogs.askBeforeExit()) {
+			Global.dbEngine.shutdown();
```

Comment:
Again, this should be a call to `UsagiApplication.shutdown`, which can call `shutdown` and `close` on the appropriate services and resources.